Lib/glyphsLib/builder/glyph.py

```diff
@@ -35,9 +35,16 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph_data):
     if last_change is not None:
         ufo_glyph.lib[GLYPHLIB_PREFIX + 'lastChange'] = to_ufo_time(last_change)
     color_index = glyph_data.color
-    if color_index is not None and color_index >= 0:
+    if color_index is not None:
         ufo_glyph.lib[GLYPHLIB_PREFIX + 'ColorIndex'] = color_index
-        ufo_glyph.lib[PUBLIC_PREFIX + 'markColor'] = GLYPHS_COLORS[color_index]
+        color_tuple = None
+        if isinstance(color_index, list):
+            color_tuple = ','.join(['{0:.4f}'.format(i/255.0)
+                                    if i in range(1, 255) else str(int(i/255)) for i in color_index])
```

Comment:
Curious how did you come up with this formula?

Reply:
A little trial and error. We know it needs to be in the range 0-255, but if we don't adjust the precision then the UFO gets a bunch of ugly values like 0.0,0.952941176471,1.0,0.00392156862745
If we set precision to 4 places we get 0.0000,0.9529,1.0000,0.0039 which is still kind of ugly. That's why the weird formula to only give 4 place precision for anything between 0 and 1, but for 0 and 1 we don't need anything so ignore it on those.